### PR TITLE
[primTorch] add meta implementation for `aten.min.dim`

### PR DESCRIPTION
**Pull request by nkaretnikov**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #96442

cc @soumith @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

### CODE REVIEW

**Comment by nkaretnikov** [comment]
See https://github.com/pytorch/pytorch/pull/88005. Added to unlock inductor tests with dyn-shapes that rely on min.

File: torch/_meta_registrations.py [hunk 1]
```diff
@@ -170,6 +170,16 @@ def meta_min(self):
     return self.new_empty(())
 
 
+@register_meta(aten.min.dim)
```
[the remark this answers is not in the document]
Comment (nkaretnikov) [reply torch/_meta_registrations.py:173]: We don't have a `min` decomp/ref AFAICT. I'm adding refs in the next commit in the stack. Note: `minimum` and `amin` are not the same as `min`.

This is symmetric with how `max` is defined. I'm only doing it to enable inductor tests that rely on `min` (part of this PR). `min` in ATen uses `structured` so cannot support `SymInt` unless you add meta (known codegen limitation).

**Review by ngimel — APPROVED** [review]

**Pull request closed without merging**